spec/models/user_spec.rb

```diff
@@ -1,4 +1,133 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
+  subject { described_class.new }
+
+  it 'user is valid' do
+    subject.first_name = 'John'
+    subject.last_name = 'Doe'
+    subject.email = 'johndoe@gmail.com'
+    subject.username = 'jdoe'
+    subject.phone_number = '0449 774 759'
+
+    expect(subject).to be_valid
+  end
+
+  # Verifying if object attribute presence is functional
+  it 'first name is present' do
+    subject.first_name = ' '
+    expect(subject).to_not be_valid
+  end
+
+  it 'last name is present' do
+    subject.last_name = ' '
+    expect(subject).to_not be_valid
+  end
+
+  it 'username is present' do
+    subject.username = ' '
+    expect(subject).to_not be_valid
+  end
+
+  it 'email is present' do
+    subject.email = ' '
+    expect(subject).to_not be_valid
+  end
+
+  it 'phone number is present' do
+    subject.phone_number = ' '
+    expect(subject).to_not be_valid
+  end
+
+  # Verifying length of object attributes
+
+  it 'first name is not too long' do
+    subject.first_name = 'a' * 51
+    expect(subject).to_not be_valid
+  end
+
+  it 'last name is not too long' do
```

Comment:
(style) write your description for what you are testing. Here we are testing that the last name **is** too long. See above context block version for how we can make this more readable.

Reply:
Ahh yeah that definitely makes sense, I'll ensure to maintain an easier to read spec file in the future, thank you!